app/src/main/java/com/example/plantsapp/presentation/ui/plants/PlantsViewModel.kt

```diff
@@ -12,31 +12,45 @@ import com.example.plantsapp.presentation.ui.utils.combineWith
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
+import timber.log.Timber
 import javax.inject.Inject
 
+@Suppress("TooGenericExceptionCaught")
 @HiltViewModel
 class PlantsViewModel @Inject constructor(
     @FirebaseQualifier private val repository: PlantsRepository
 ) : ViewModel() {
 
-    private val allPlants: MutableLiveData<List<Plant>> = MutableLiveData(emptyList())
+    private val allPlants: MutableLiveData<List<Plant>> = MutableLiveData()
     private val filter: MutableLiveData<String> = MutableLiveData("")
     val filteredPlants: LiveData<List<Plant>> = allPlants.combineWith(filter) { plants, filter ->
-        plants!!.filter {
+        plants?.filter {
```

Comment:
Possibly it is good candidate to use sealed class too

Reply:
Also here you give ability to set null here , so maybe type should be corresponding